src/components/StudyPane/InfoPane/Motif/RootBlock.tsx

```diff
@@ -1,62 +1,49 @@
 import React, { useState, useEffect, useContext } from 'react';
 
 import { DEFAULT_COLOR_FILL, DEFAULT_BORDER_COLOR, DEFAULT_TEXT_COLOR, FormatContext } from '../../index';
-import { ColorActionType } from "@/lib/types";
+import { ColorActionType, ColorType } from "@/lib/types";
 import { HebWord } from '@/lib/data';
+import { updateWordColor } from '@/lib/actions';
 
 export const RootBlock = ({
-  id, count, descendants
+    id, count, descendants
 }: {
-  id: number,
-  count: number,
-  descendants: HebWord[]
+    id: number,
+    count: number,
+    descendants: HebWord[]
 }) => {
 
-  const { ctxRootsColorMap, ctxColorAction, ctxSelectedColor, ctxSelectedHebWords, ctxSetNumSelectedWords, ctxSetSelectedHebWords } = useContext(FormatContext)
-  
-  let defaultColorFill = descendants[0].colorFill || DEFAULT_COLOR_FILL;
-  let defaultBorderColor = descendants[0].borderColor || DEFAULT_BORDER_COLOR;
-  let defaultTextColor = descendants[0].textColor || DEFAULT_TEXT_COLOR;
-
-  const rootBlockColor = ctxRootsColorMap.get(descendants[0].strongNumber);
-  if (rootBlockColor) {
-    defaultColorFill = rootBlockColor.colorFill;
-    defaultTextColor = rootBlockColor.textColor;
-  }
+  const { ctxStudyId, ctxRootsColorMap, ctxColorAction, ctxSelectedColor, ctxSelectedHebWords,
+    ctxSetNumSelectedWords, ctxSetSelectedHebWords, ctxSetRootsColorMap } = useContext(FormatContext)
 
-  descendants.forEach((dsd) => {
-    if (dsd.colorFill !== defaultColorFill) { defaultColorFill = DEFAULT_COLOR_FILL; }
-    if (dsd.borderColor !== defaultBorderColor) { defaultBorderColor = DEFAULT_BORDER_COLOR; }
-    if (dsd.textColor !== defaultTextColor) { defaultTextColor = DEFAULT_TEXT_COLOR; }
-  });
-
-  const [colorFillLocal, setColorFillLocal] = useState(defaultColorFill);
-  const [borderColorLocal, setBorderColorLocal] = useState(defaultBorderColor);
-  const [textColorLocal, setTextColorLocal] = useState(defaultTextColor);
+  const [colorFillLocal, setColorFillLocal] = useState(descendants[0].colorFill? descendants[0].colorFill: DEFAULT_COLOR_FILL);
+  const [borderColorLocal, setBorderColorLocal] = useState(descendants[0].borderColor? descendants[0].colorFill: DEFAULT_BORDER_COLOR);
+  const [textColorLocal, setTextColorLocal] = useState(descendants[0].textColor? descendants[0].colorFill: DEFAULT_TEXT_COLOR);
   const [selected, setSelected] = useState(false);
 
+  const matchExpandedWordsColorScheme = () => {
```

Comment:
We don't need this change

Reply:
It should look like this now: 
```typescript
const matchFillColor = () => {
    let match = descendants.every((dsd) => {
      return !dsd.colorFill || dsd.colorFill === descendants[0].colorFill;
    });
    return match;
  }

  const matchTextColor = () => {
    let match = descendants.every((dsd) => {
      return !dsd.textColor || dsd.textColor === descendants[0].textColor;
    });
    return match;
  }

  const matchBorderColor = () => {
    let match = descendants.every((dsd) => {
      return !dsd.borderColor || dsd.borderColor === descendants[0].borderColor;
    });
    return match;
  }
```

I need just some utility functions to match all descendants by their colour scheme and broke them down into separate functions. The behaviour it enables is that whenever the user selects all the descendants belonging under one root, changing colours does not immediately clear their colour scheme if they happen to not be a match. The descendants keep their colour as is until it is changed in a certain category (i.e. border/ fill/ text colour)